### ISSUE TITLE
Key combinations

### ISSUE THREAD

**Issue by Lange** [Lange/node-elgato-stream-deck#9]
Being able to register a callback for when an arbitrary combination of keys is pressed would be useful. Here's an example of what an API for this _might_ look like (feedback welcome):
```js
streamDeck.registerKeyCombo([4, 8, 11], () => {
    // do something when keys 4, 8, and 11 are all down at the same time
});
```

**Comment by ProbablePrime** [comment]
Woops accidental assignment

**Comment by Lange** [comment]
I haven't thought of a better API for this, so I'd be okay with someone going ahead and implementing the API described in the original post and opening a pull request for it.

**Comment by Lange** [comment]
Actually, maybe it should look like this:
```js
const keyCombo = streamDeck.registerKeyCombo([4, 8, 11]); // returns an EventEmitter
keyCombo.on('down', keys => {
    console.log(keys); // => [4, 8, 11]
});
keyCombo.on('up', keys => {
    console.log(keys); // => [4, 8, 11]
});
keyCombo.unregister(); // Removes all listeners.
```

**Comment by Lange** [comment]
Oh interesting, you mean registering key combos which use some keys on the Stream Deck and some keys on a keyboard, in a single combo?

**Comment by pkej** [comment]
Yes. But by leveraging https://github.com/agirorn/node-hid-stream and use the HidKeyboardPacket as an expanded return on event up and down, expanding on the current keyindex returned today. The HidKeyboardPacket can already return multiple key presses. I guess it could be expanded to include multiple keyindex values in the same way as keycodes. Node-hid-stream depends on node-hid so a good match.

**Comment by pkej** [comment]
A further expansion would be to create a buffer for all the graphics on the keyboard and store these for “layers” that could be loaded on modifier key presses. So the stream deck will show one default layer initially, but for each layer applied to a modifier key combo it would change to show what the modifier would do. By stitching all the icons assigned to each physical key programmatically it would be pretty neat.

**Comment by Lange** [comment]
I think for now, I don't want to include the ability to interact with other HID devices in this library. It seems like scope creep and would make this library harder to maintain.

On the topic of "layers", yes I agree that it'd be good for this library to expose some methods which assist users in implementing a multi-layered set of controls for their stream deck. I've not yet put any thought into what that API might look like. There's nothing stopping users from implementing this in their application code though, and doesn't necessarily require that we implement it in the library.

**Comment by pkej** [comment]
I’ll be hacking tomorrow, after redeploying Windows and reinstalling node. I can see how feature creep would be bad. I’m thinking layers etc would probably be great as an GUI-config. I’ve got a lot of stuff lying around, x-keys, strategic commander, midi controllers, tactical keyboard, shuttle pro, ad nauseam, and also came across http://osc.ammd.net which is snappy on my iPad. So I think I need to model the different ways of layering for different devices.
